Code/GraphMol/MarvinParse/MarvinParser.cpp

```diff
@@ -0,0 +1,1110 @@
+            pos.x = atomPtr->x2;
+            pos.y = atomPtr->y2;
+          } else {
+            pos.x = 0.0;
+            pos.y = 0.0;
+          }
+          pos.z = 0.0;
+
+          conf->setAtomPos(aid, pos);
+        }
+
+        // also collect the stereo groups here
+
+        if (atomPtr->mrvStereoGroup != "") {
+          RDKit::StereoGroupType groupType;
+          int groupNumber;
+
+          // get the group parts
+          std::string temp =
+              boost::algorithm::to_lower_copy(atomPtr->mrvStereoGroup);
+          if (temp == "abs") {
+            groupType = RDKit::StereoGroupType::STEREO_ABSOLUTE;
+            groupNumber = (-1);
+          } else if (boost::starts_with(temp, "and")) {
+            groupType = RDKit::StereoGroupType::STEREO_AND;
+            if (!getCleanInt(temp.substr(3), groupNumber)) {
+              throw FileParseException(
+                  "Group Number must be an integer in a stereo group AND# in a MRV file");
+            }
+          } else if (boost::starts_with(temp, "or")) {
+            groupType = RDKit::StereoGroupType::STEREO_OR;
+            if (!getCleanInt(temp.substr(2), groupNumber)) {
+              throw FileParseException(
+                  "Group Number must be an integer in a stereo group OR# in a MRV file");
+            }
+          } else {
+            throw FileParseException("Unrecognized group definition");
+          }
+
+          // see if the group already exists
+
+          MarvinStereoGroup *marvinStereoGroup;
+          auto groupIter =
+              find_if(stereoGroups.begin(), stereoGroups.end(),
+                      [groupType, groupNumber](const MarvinStereoGroup *arg) {
+                        return arg->groupNumber == groupNumber &&
+                               arg->groupType == groupType;
+                      });
+          if (groupIter != stereoGroups.end()) {
+            marvinStereoGroup = *groupIter;
+          } else {
+            marvinStereoGroup = new MarvinStereoGroup(groupType, groupNumber);
+            stereoGroups.push_back(marvinStereoGroup);
+          }
+
+          // add this atom to the group
+
+          marvinStereoGroup->atoms.push_back(
+              (unsigned int)marvinMol->getAtomIndex(atomPtr->id));
+        }
+      }
+
+      if (conf != nullptr) {
+        mol->addConformer(conf, true);
+        conf = nullptr;  // conf now owned by mol
+      }
+
+      // set the bonds
+
+      bool chiralityPossible = false;
+
+      for (auto bondPtr : marvinMol->bonds) {
+        molBondFromMarvinBond(bondPtr, marvinMol, mol, chiralityPossible);
+      }
+
+      //  add the stereo groups
+
+      std::vector<StereoGroup> groups;
+
+      for (auto groupPtr : stereoGroups) {
+        std::vector<Atom *> atoms;
+        for (auto atomPtr : groupPtr->atoms) {
+          atoms.push_back(mol->getAtomWithIdx(atomPtr));
+        }
+
+        groups.emplace_back(groupPtr->groupType, std::move(atoms));
```

Comment:
Stereo groups can now track what their original ID was (this was added in #6560). Might as well go ahead and pass the original group number along when you construct the group.

Reply:
OK